src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -113,6 +115,29 @@ public override int Execute()
             }
             else
             {
+                var globaljsonPath = SdkDirectoryWorkloadManifestProvider.GetGlobalJsonPath(Environment.CurrentDirectory);
+                var workloadVersion = SdkDirectoryWorkloadManifestProvider.GlobalJsonReader.GetWorkloadVersionFromGlobalJson(globaljsonPath);
+                if (workloadVersion is not null)
+                {
+                    if (string.IsNullOrWhiteSpace(_workloadSetVersion))
+                    {
+                        _workloadSetVersion = workloadVersion;
+                    }
+                    else if (!workloadVersion.Equals(_workloadSetVersion))
+                    {
+                        throw new Exception(string.Format(LocalizableStrings.CannotSpecifyVersionOnCommandLineAndInGlobalJson));
+                    }
+                }
+                else if (string.IsNullOrWhiteSpace(_workloadSetVersion))
+                {
+                    var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                    if (File.Exists(installStateFilePath))
+                    {
+                        var installStateContents = InstallStateContents.FromPath(installStateFilePath);
+                        _workloadSetVersion = installStateContents.WorkloadVersion;
+                    }
```

Comment:
You were right! And more importantly, with how this case was handled (in the constructor), it would interpret something from the install state as a command line argument early, which means it would error if the install state and global.json disagreed. Good catch!